src/api.v2/helpers/s3/signedUrl.js

```diff
@@ -38,31 +36,33 @@ const createMultipartUpload = async (params, size) => {
 
   const { UploadId } = await s3.createMultipartUpload(params).promise();
 
-  const baseParams = {
-    ...params,
-    UploadId,
+  return {
+    uploadId: UploadId,
+    bucket: params.Bucket,
+    key: params.Key,
   };
+};
 
-  const promises = [];
-  const parts = Math.ceil(size / FILE_CHUNK_SIZE);
-
-  for (let i = 0; i < parts; i += 1) {
-    promises.push(
-      s3.getSignedUrlPromise('uploadPart', {
-        ...baseParams,
-        PartNumber: i + 1,
-      }),
-    );
-  }
```

Comment:
Now we don't try to predict how many parts we'll send, we just send them whenever the read/compress stream makes them available and request a signed url for each of them. This request is managed with the new endpoint in `uploadController`